app/Services/Model/Imp/ScheduleService.php

```diff
@@ -403,4 +424,68 @@ public function publishAll(string $source, int $summit_id, array $payload, ?Filt
 
         return $schedule;
     }
+
+    /**
+     * @inheritDoc
+     */
+    public function addReview(Summit $summit, string $source, int $track_id, array $payload): SummitProposedSchedule
+    {
+        return $this->tx_service->transaction(function () use ($summit, $source, $track_id, $payload) {
+
+            $former_lock = $this->schedule_lock_repository->getBySummitAndTrackId($summit->getId(), $track_id);
+
+            if (!is_null($former_lock)) {
+                SubmitForReviewEmail::dispatch($former_lock);
+                throw new ValidationException("this track already has a review submission");
+            }
+
+            $schedule = $this->schedule_repository->getBySourceAndSummitId($source, $summit->getId());
+
+            if (!$schedule instanceof SummitProposedSchedule) {
+                throw new EntityNotFoundException("can not find a proposed schedule for summit id {$summit->getId()} and source {$source}");
+            }
+            $track = $summit->getPresentationCategory($track_id);
+
+            if (!$track instanceof PresentationCategory) {
+                throw new EntityNotFoundException("can not find a presentation category with id {$track_id} for summit id {$summit->getId()}");
+            }
+            $member = ResourceServerContext::getCurrentUser(false);
+            $track_chair = $summit->getTrackChairByMember($member);
+
+            if (!$track_chair instanceof SummitTrackChair) {
+                throw new EntityNotFoundException("can not find a track chair for current member {$member->getId()}");
+            }
+            $lock = new SummitProposedScheduleLock();
+
+            if (isset($payload['message']))
+                $lock->setReason($payload['message']);
+
+            $lock->setTrack($track);
+            $lock->setCreatedBy($track_chair);
+
+            $schedule->addProposedScheduleLock($lock);
+            return $schedule;
+        });
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function removeReview(Summit $summit, string $source, int $track_id): SummitProposedSchedule
```

Comment:
u need here the optional message from admin as a param

Reply:
done